rest/data/task_test.go

```diff
@@ -81,6 +82,67 @@ func (s *TaskConnectorFetchByIdSuite) TestFindByIdAndExecution() {
 	}
 }
 
+func (s *TaskConnectorFetchByIdSuite) TestFindByVersion() {
+	s.Require().NoError(db.ClearCollections(task.Collection, task.OldCollection, annotations.Collection))
+	task_known := &task.Task{
+		Id:        "task_known",
+		Execution: 0,
+		Version:   "version_known",
+	}
+	task_not_known := &task.Task{
+		Id:        "task_not_known",
+		Execution: 0,
+		Version:   "version_not_known",
+		Status:    evergreen.TaskFailed,
+	}
+	task_no_annotation := &task.Task{
+		Id:        "task_no_annotation",
+		Execution: 0,
+		Version:   "version_no_annotation",
+		Status:    evergreen.TaskFailed,
+	}
+	task_with_empty_issues := &task.Task{
+		Id:        "task_with_empty_issues",
+		Execution: 0,
+		Version:   "version_with_empty_issues",
+		Status:    evergreen.TaskFailed,
+	}
+	s.NoError(task_known.Insert())
+	s.NoError(task_not_known.Insert())
+	s.NoError(task_no_annotation.Insert())
+	s.NoError(task_with_empty_issues.Insert())
+
+	issue := annotations.IssueLink{URL: "https://issuelink.com", IssueKey: "EVG-1234", Source: &annotations.Source{Author: "chaya.malik"}}
+
+	a_with_issue := annotations.TaskAnnotation{TaskId: "task_known", TaskExecution: 0, Issues: []annotations.IssueLink{issue}}
```

Comment:
Can we also add a test with multiple annotations for different executions to verify that we're querying execution with taskId correctly?

Reply:
done